src/GitVersionCore.Tests/BuildServers/VsoAgentTests.cs

```diff
@@ -1,16 +1,35 @@
-﻿using GitVersion;
+﻿using System;
+using GitVersion;
+using GitVersionCore.Tests;
 using NUnit.Framework;
 using Shouldly;
 
 [TestFixture]
 public class VsoAgentTests
 {
+
+    string key = "BUILD_BUILDNUMBER";
+
+    private void SetEnvironmentVariableForTest()
+    {
+        Environment.SetEnvironmentVariable(key, "Some Build_Value $(GitVersion_FullSemVer)", EnvironmentVariableTarget.Process);
+    }
+
+    private void ClearEnvironmentVariableForTest()
+    {
+        Environment.SetEnvironmentVariable(key, null, EnvironmentVariableTarget.Process);
+    }
+
     [Test]
     public void Develop_branch()
     {
+        SetEnvironmentVariableForTest();
```

Comment:
Wouldn't it be better to call SetEnvironmentVariableForTest in a Setup method (since they are also around all the time in the build server)?